### ISSUE TITLE
Allow either guid or url in the first column of the CSV files

### ISSUE THREAD

**Issue by timhodson** [jmunzer/bulk_tools#18]
All List Item Report from Talis Aspire contains a URL.  We could remove a pre-processing step which is imposed on users by having the code allow and detect wither a list item URL or a bare GUID.

i.e. except input in the first column that looks like any of these:

```
http://tenancy.rl.talis.com/items/abcd1234-abcd-efgh-adbc5432
http://tenancy.rl.talis.com/items/abcd1234-abcd-efgh-adbc5432.html
http://tenancy.rl.talis.com/items/abcd1234-abcd-efgh-adbc5432.html?some=params
abcd1234-abcd-efgh-adbc5432
```

This could be done in the while loop that reads the line, having a function that checks the first column's value and detects whether it needs to extract a guid or not.

The same could be done for lists and any other Talis identifiers handled by the API which could exist in those different forms.

Basically this is about making the scripts more forgiving.

**Comment by jmunzer** [comment]
Ah, I did think about this a few days ago.. happy to build this out. I was thinking about how best to trim though as tenancy prefixes will vary in size and I wasn't sure if from the right, all GUIDs were/will always be equal length?

Perhaps target /items/ and grab everything to the right of that?

I'll look at building something out early next week perhaps

**Comment by timhodson** [comment]
Yep, I would split the string on `/items/` using the php 'explode()` function, then the second element of the array you get back will contain the id plus maybe some other stuff.
I would then explode again that second element using a `.` character, and the first element in that next array will be a guid.

Might then want to also double check that we really have got a guid... https://stackoverflow.com/questions/1253373/php-check-for-valid-guid 

If you plan to work on this issue, you can 'self assign' the issue so that folks can see whether something might be being worked on. :)

**Comment by jmunzer** [comment]
gotchya, thanks for the tips!
